src/main/java/racingcar/service/GameService.java

```diff
@@ -0,0 +1,55 @@
+package racingcar.service;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import racingcar.domain.RacingCar;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GameService {
+    private static final int MOVE_NUMBER = 4;
+
+    private List<RacingCar> racingCars;
+
+    public GameService(String carNames) {
+        racingCars = new ArrayList<>();
+        initializeRacingCars(carNames);
+    }
+
+    private void initializeRacingCars(String carNames) {
+        racingCars = Arrays.stream(carNames.split(","))
+                .map(name -> new RacingCar(name.trim()))
+                .collect(Collectors.toList());
+    }
+
+    private boolean canMove() {
+        int pickedNumber = Randoms.pickNumberInRange(0, 9);
+        return pickedNumber >= MOVE_NUMBER;
+    }
+
+    public List<RacingCar> game() {
+        racingCars.stream()
+                .filter(car -> canMove())
+                .forEach(RacingCar::moveForward);
+
+        return racingCars;
+    }
+
+    public List<String> findWinner() {
+        int maxDistance = findMaxDistance();
+
+        return racingCars.stream()
+                .filter(car -> car.getDistance() == maxDistance)
```

Comment:
위의 코드에서 객체지향을 잘 활용하기 위해 getter()를 통해 값을 비교하는 대신, RacingCar에게 메시지를 보내서 값을 비교하는 것은 어떨까요? 예를 들어, 다음과 같이 수정할 수 있습니다:

```suggestion
                .filter(car -> car.isMaxDistance(maxDistance))
```

Reply:
getter()가 아니라 요청 메시지를 통해서 비교하는 방법도 있었네요.
감사합니다!